src/rotation.zig

```diff
@@ -1540,6 +1540,90 @@ pub const RotationModes = struct {
 			currentData.data = 0;
 		}
 	};
+	pub const TexturePile = struct {
+		pub const id: []const u8 = "texturePile";
+		var rotatedModels: std.StringHashMap(ModelIndex) = undefined;
+		var blockToStateCountMap: std.AutoHashMapUnmanaged(u16, u16) = undefined;
+
+		fn init() void {
+			rotatedModels = .init(main.globalAllocator.allocator);
+			blockToStateCountMap = .{};
+		}
+
+		fn deinit() void {
+			rotatedModels.deinit();
+			blockToStateCountMap.deinit(main.globalAllocator.allocator);
+		}
+
+		fn reset() void {
+			rotatedModels.clearRetainingCapacity();
+			blockToStateCountMap.clearRetainingCapacity();
+		}
+
+		fn transform(quad: *main.models.QuadInfo, data: u16) void {
+			quad.textureSlot = data%16;
+		}
+
+		pub fn createBlockModel(blockId: []const u8, zon: ZonElement) ModelIndex {
+			const modelId = zon.get([]const u8, "model", "cubyz:cube");
+			const stateCount = zon.get(u16, "states", 2);
+			std.debug.assert(stateCount > 1); // There is not enough information to send meaningful log.
```

Comment:
There is the block id. Is that not good enough?
Also we shouldn't crash the game when an addon creator makes a mistake (especially given that asserts only trigger in safe builds, what if the addon creator only tested a ReleaseFast build?)

Reply:
Comments easily get outdated don't they. Ofc with current code it's enough information.